### PR TITLE
Add ability to set various cilium flags through CLI

### PR DESCRIPTION
**Pull request by olemarkus**
Needed for a new e2e cilium test

### CODE REVIEW

**Comment by olemarkus** [comment]
Want to get `etcdManaged` in as well.

File: pkg/commands/set_cluster_test.go [hunk 1]
```diff
@@ -70,3 +70,26 @@ func TestSetClusterFields(t *testing.T) {
...
+			Kubelet: &kops.KubeletConfigSpec{},
+		},
+	}
+
+	var igs []*kops.InstanceGroup
+
+	err := SetClusterFields(fields, cluster, igs)
+	if err != nil {
+		t.Errorf("unexpected error from setClusterFields %v: %v", cluster, err)
+	}
```
Comment (rifelpet) [inline pkg/commands/set_cluster_test.go:94]: can you assert that the fields are correctly set in the `cluster` object?

```
if cluster.Spec.Networking.Cilium.Ipam != "eni" {
	t.Errorf("...")
}
```
etc

**Comment by rifelpet** [comment]
In order to test the etcdManaged under e2e tests we'll have to figure out how to add an additional etcdCluster named `cilium` I'm struggling to come up with a simple solution for that...

File: pkg/commands/set_cluster_test.go (hunk 1 above)
Comment (olemarkus) [reply pkg/commands/set_cluster_test.go:94]: Yep. Just since there are pointers involved, I need to properly demarshal the spec to yaml, I guess

**Comment by olemarkus** [comment]
Yeah. I am wondering if the only feasible option is a proper cli flag.

**Comment by rifelpet** [comment]
/lgtm

Thoughts on the new cli flag @justinsb or anyone else?
/assign @justinsb

**Comment by olemarkus** [comment]
How far away is this one from approval now?

**Comment by hakman** [comment]
One thought here: instead of a new flag would it be a good idea to reuse the networking one, like `--networking=cilium-etcd`? I think there are similar flags for Flannel.

**Comment by olemarkus** [comment]
I am not a big fan of adding more to `--networking`. `--networking` is a significant flag and adding something that is just to work around limitation of `kubetest` isn't that user friendly I think.

**Comment by hakman** [comment]
I am debating here that it may be the easiest way to get people started with Cilium + Etcd. It is not just good for kubetest, but would generate the correct config automatically, instead of doing some steps prone to user error.
From what I see this seems to be the way to activate a particular config for a certain network plugin.

**Comment by olemarkus** [comment]
It may be an idea to rethink this a bit. Nodeport and ENI is the most important features to test. For networking values, these two also makes more sense perhaps. 

etcd is less important to test. Also, if etcd-manager works well, it may also be an idea to switch that on by default, which means that cli flag is perhaps not that useful.

**Comment by hakman** [comment]
My take on this is that you should have a default as close as possible to upstream that will work without issues to create smaller test clusters. Think about adding various `--networking=cilium-<type>` flags to create more complex setups.
In any case, this is mostly a nit. Most work on this is really ok. Let's see also if @johngmyers has some feedback.

**Comment by johngmyers** [comment]
There appears to be precedent in `flannel-vxlan` vs. `flannel-udp` for putting it in `--networking`.  Unfortunately, Cilium appears to have three independent modes, so how would one say one wants Etcd and Nodeport, but not ENI?

We could do something like `--networking=cilium=etcd,nodeport` but that's getting unwieldy. The only other idea I have is a `--networking-option` flag of type `[]string`.

**Comment by olemarkus** [comment]
With #9425 the more complex option will be on by default. ENI mode could be interesting to test still though.

**Comment by olemarkus** [comment]
Rebased to #9490

**Comment by hakman** [comment]
@rifelpet I think this is ready.
/lgtm

**Comment by rifelpet** [comment]
ok, i took a final pass and it looks good to me and is pretty straight forward 👍 thanks for sticking with it through all the rebases.

/approve

**Pull request merged**